lib/opensearch-aws-sigv4.rb

```diff
@@ -37,7 +37,42 @@ module Aws
     #
     # @attr [Aws::Sigv4::Signer] sigv4_signer Signer used to sign every request
     class Sigv4Client < ::OpenSearch::Client
-      attr_accessor :sigv4_signer
+      # <tt>Aws::Sigv4::Signer</tt> wrapper for <tt>::OpenSearch::Transport::Client</tt>.
+      class RequestSigner
+        def initialize(sigv4_signer, sigv4_debug)
+          @sigv4_signer = sigv4_signer
+          @sigv4_debug = sigv4_debug
```

Comment:
Since user of this gem can provide their own logger object, we also dont need the `debug` flag anymore. Simply log the signature in the debug stream (`logger&.debug(...)`) and let the user worry about how to view/store this info since they control the logger object now.